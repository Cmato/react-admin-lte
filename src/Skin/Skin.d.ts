import * as React from 'react';

export interface SkinProps {
  color: SkinColor;
}

export type SkinColor = 'blue' | 'black' | 'purple' | 'green' | 'red' |
  'yellow' | 'blue-light' | 'black-light' | 'purple-light' | 'green-light' |
  'red-light' | 'yellow-light';

declare const Skin: React.ComponentType<SkinProps>;

export default Skin;
