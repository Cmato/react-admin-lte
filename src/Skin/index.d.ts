export { default } from './Skin';
export * from './Skin';
