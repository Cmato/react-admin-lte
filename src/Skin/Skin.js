import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const Skin = ({
  children,
  className: classNameProp,
  color,
  ...other
}) => {
  const className = classNames({
    [`skin-${color}`]: true
  }, classNameProp);

  return (
    <div className={className} {...other}>
      {children}
    </div>
  );
};

Skin.propTypes = {
  /**
   * The content of the skin.
   */
  children: PropTypes.node.isRequired,
  /**
   * @ignore
   */
  className: PropTypes.string,
  /**
   * Color of the skin to display.
   */
  color: PropTypes.oneOf([
    'blue', 'black', 'purple', 'green', 'red', 'yellow', 'blue-light', 'black-light',
    'purple-light', 'green-light', 'red-light', 'yellow-light'
  ]).isRequired
};

export default Skin;
