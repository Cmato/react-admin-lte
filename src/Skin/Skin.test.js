import React from 'react';
import { shallow } from 'enzyme';
import Skin from './Skin';

it('renders blue', () => {
  expect(shallow((
    <Skin color="blue" className="best-skin">
      <div />
    </Skin>
  ))).toMatchSnapshot();
});
