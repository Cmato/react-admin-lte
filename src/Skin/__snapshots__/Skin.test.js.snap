// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders blue 1`] = `
<div
  className="skin-blue best-skin"
>
  <div />
</div>
`;
